public_html/requests/worksheetFunctions.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+$include_path = get_include_path();
+include_once $include_path . '/includes/db_functions.php';
+include_once $include_path . '/includes/session_functions.php';
+include_once $include_path . '/public_html/classes/AllClasses.php';
+include_once $include_path . '/public_html/requests/core.php';
+
+$requestType = filter_input(INPUT_POST,'type',FILTER_SANITIZE_STRING);
+$vid = filter_input(INPUT_POST,'vid',FILTER_SANITIZE_NUMBER_INT);
+$userid = filter_input(INPUT_POST,'userid',FILTER_SANITIZE_NUMBER_INT);
+$userval = base64_decode(filter_input(INPUT_POST,'userval',FILTER_SANITIZE_STRING));
+
+$role = validateRequest($userid, $userval);
+if(!$role){
+    failRequest("There was a problem validating your request");
+}
+
+switch ($requestType){
+    case "DELETE":
+        if(!authoriseUserRoles($role, ["SUPER_USER", "STAFF"])){
```

Comment:
Maybe delete should be super user only/teacher can delete a worksheet they created